### ISSUE TITLE
Execution order changes between UserObjects when AuxKernel is included

### ISSUE THREAD

**Issue by SudiptaBiswas** [idaholab/moose#7884]
### Description of the enhancement or error report
Usually `GeneralUserObjects` are executed after all other UserObjects. This execution order changes when there exist an `AuxKernel` that depends on the `GeneralUserObject`.

### Rationale for the enhancement or information for reproducing the error
Let's say, I have a`GeneralUserObject` that depends on an `ElementUserObject` and an `Auxkernel` that depends on the `GeneralUserObject`. The `ElementUserObject` has no relation with the `AuxKernel` here. 

Now, since the `AuxKernel` depends on `GeneralUserObject`, the said UO is initialized at `PRE-AUX`. However, the `ElementUserObject` is not initialized at that point (its done `POST-AUX`). This results in segfault/incorrect execution. 

ping @aeslaughter @dschwen

**Comment by SudiptaBiswas** [comment]
8e0cc25652b07d1f6d742370792c54b5c25a0a31 shows how inclusion of `AuxKernel` results in segfault.

**Comment by dschwen** [comment]
If that is the problem you could try to force the Element UO into PRE-AUX as well using a dummy AuxKernel.

**Comment by YaqiWang** [comment]
This is a long-stand issue. One solution is to make you `AuxKernel` depend on also the `ElementUserObject` although it does not depend on it directly. This way you can spin both user objects into the pre-aux group.

**Comment by SudiptaBiswas** [comment]
Yeah, that would be a temporary fix.

**Comment by permcody** [comment]
Yeah we probably won't fix this anytime soon. Arbitrary dependency
resolution is something we all want but it's a lot of work to get right.
